src/ComplexScalars/UriTrait.php

```diff
@@ -72,4 +72,83 @@ public function toNative()
     {
         return $this->uri->__toString();
     }
+
+    /**
+     * @return UriInterface
+     */
+    public function getPsr7Uri()
```

Comment:
Can we have PHP 7 return types pretty please?

Reply:
Yep - sorry, too much time on another codebase!